torch/optim/lr_scheduler.py

```diff
@@ -303,3 +314,74 @@ def _init_is_better(self, mode, threshold, threshold_mode):
         else:  # mode == 'max' and epsilon_mode == 'abs':
             self.is_better = lambda a, best: a > best + threshold
             self.mode_worse = -float('Inf')
+
+
+class ReduceLROnPlateauWithBacktrack(ReduceLROnPlateau):
```

Comment:
I think it would be better to add a `backtrack` argument to `ReduceLROnPlateau`

Reply:
Not sure about this -- we would need two more parameters for backtracking.

Reply:
If you need a file name make it `backtrack_filename` which enables backtracking if set, and is None otherwise (which means no backtracking)